### ISSUE TITLE
IllegalStateException: Expected a string but was NULL

### ISSUE THREAD

**Issue by grigoryk** [mozilla-mobile/android-components#6437]
https://sentry.prod.mozaws.net/operations/fenix-fennec-beta/issues/7666560/

```
IllegalStateException: Expected a string but was NULL
    at android.util.JsonReader.nextString
    at mozilla.components.support.migration.session.StreamingSessionStoreParser.parseWindows(StreamingSessionStoreParser.kt:30)
    at mozilla.components.support.migration.session.StreamingSessionStoreParser.parse$support_migration_release(StreamingSessionStoreParser.kt:8)
    at mozilla.components.support.migration.FennecSessionMigration.migrate(FennecSessionMigration.kt:12)
    at mozilla.components.support.migration.FennecMigrator.migrateOpenTabs(FennecMigrator.kt:14)
...
(6 additional frame(s) were not displayed)

[INFO] Expected a string but was NULL
```

**Comment by grigoryk** [comment]
Sentry issue: [FENIX-FENNEC-BETA-2M](https://sentry.prod.mozaws.net/operations/fenix-fennec-beta/issues/7666559/)

**Comment by pocmo** [comment]
Unfortunately the stacktrace is not showing the exact line. The only `String` that we read and where we do no explicit `null` check is the `url` of an tab entry.

**Comment by pocmo** [comment]
Fennec doesn't treat the URL as optional either:
https://searchfox.org/mozilla-esr68/source/mobile/android/base/java/org/mozilla/gecko/SessionParser.java#108

Maybe we should not change that for now and monitor (only one user affected). It may be correct to fail (it's not a crash) and potentially try `sessionstore.bak` (if this happened with `sessionstore.js`).

**Comment by pocmo** [comment]
6 users so far.

**Comment by pocmo** [comment]
Closing: I think we should not make a code change here.

Reasoning: From the data we collect we do not know whether we have failed to read the session store or its backup. If we failed reading the store here then we have moved on to read the backup, which is okay and what Fennec would do too ([code](https://searchfox.org/mozilla-esr68/source/mobile/android/base/java/org/mozilla/gecko/SessionParser.java#108)). If we ignore this error and continue to read the "broken" store then we may end up restoring something that Fennec would not have restored either and that may be more problematic. The best solution here would be to have more verbose logging and figure out what exactly went wrong and if we could recover. But that is not really feasible without a test audience to migrate. I wouldn't want to start doing this on the release population. According to Sentry this has affected 168 beta users, which is a low number and may make it unlikely to even hit it in an early production rollout stage (e.g. 1%) to adapt early on.
